src/ls.js

```diff
@@ -36,91 +37,24 @@ function _ls(options, paths) {
     options.all = true;
   }
 
-  if (!paths)
-    paths = ['.'];
-  else if (typeof paths === 'object')
-    paths = paths; // assume array
-  else if (typeof paths === 'string')
-    paths = [].slice.call(arguments, 1);
+  if (typeof paths === 'string') paths = [].slice.call(arguments, 1);
+  paths = paths || [];
 
-  var list = [];
-
-  // Conditionally pushes file to list - returns true if pushed, false otherwise
-  // (e.g. prevents hidden files to be included unless explicitly told so)
-  function pushFile(file, query) {
-    // hidden file?
-    if (path.basename(file)[0] === '.') {
-      // not explicitly asking for hidden files?
-      if (!options.all && !(path.basename(query)[0] === '.' && path.basename(query).length > 1))
-        return false;
-    }
+  paths = paths.map(function (path) {
+    path += (options.recursive ? '/**/*.*' : '/*.*');
```

Comment:
My concern is that on unix you can do:

```Bash
$ touch a
$ touch b
$ touch c.txt
$ ls * # all files
a   b   c.txt
$ ls *.* # only files with dots in the name
c.txt
```

And it looks like this would not support that behavior. If the code does perform as expected, let me know.